caddyhttp/httpserver/server.go

```diff
@@ -265,6 +265,17 @@ func (s *Server) serveHTTP(w http.ResponseWriter, r *http.Request) (int, error)
 		}
 	}
 
+	// Apply the path-based request body size limit
+	// The error returned by MaxBytesReader is meant to be handled
+	// by whichever middleware/plugin that receives it when calling
+	// .Read() or a similar method on the request body
+	for _, pathlimit := range vhost.MaxRequestBodySizes {
+		if Path(r.URL.Path).Matches(pathlimit.Path) {
+			r.Body = http.MaxBytesReader(w, r.Body, pathlimit.Limit)
```

Comment:
I've heard this debated before, and so I'm not sure if that's true -- but I definitely would not oppose to a nil check here!